pkg/search/postgres/common.go

```diff
@@ -589,6 +597,64 @@ func RunGetManyQueryForSchema(ctx context.Context, schema *walker.Schema, q *v1.
 	return results, nil
 }
 
+// RunCursorQueryForSchema creates a cursor against the database
+func RunCursorQueryForSchema(ctx context.Context, schema *walker.Schema, q *v1.Query, db *pgxpool.Pool) (fetcher func(n int) ([][]byte, error), closer func(), err error) {
+	query, err := standardizeQueryAndPopulatePath(q, schema, GET)
+	if err != nil {
+		return nil, nil, err
+	}
+	if query == nil {
+		return nil, nil, errox.InvalidArgs.New("empty query")
+	}
+
+	queryStr := query.AsSQL()
+
+	conn, err := db.Acquire(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	closer = func() {
+		conn.Release()
+	}
+
+	tx, err := conn.Begin(ctx)
+	if err != nil {
+		closer()
+		return nil, nil, err
+	}
+	closer = func() {
+		if err := tx.Rollback(ctx); err != nil {
+			log.Errorf("error rolling back cursor transaction: %v", err)
+		}
+		conn.Release()
+	}
+
+	cursor := random.GenerateString(16, random.CaseInsensitiveAlpha)
+	_, err = tx.Exec(ctx, fmt.Sprintf("DECLARE %s CURSOR FOR %s", cursor, queryStr), query.Data...)
+	if err != nil {
+		closer()
+		return nil, nil, err
+	}
+
+	return func(n int) ([][]byte, error) {
+		rows, err := tx.Query(ctx, fmt.Sprintf("FETCH %d FROM %s", n, cursor))
```

Comment:
nit: should use sprintf here or pg template to handle escaping?

Reply:
actually only can template in the where